etc/docker/odh/setup-elyra-metadata.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Assumptions are existing kubeflow installation is in the kubeflow namespace
+DEFAULT_RUNTIME_FILE=$(jupyter --data-dir)/metadata/runtimes/my_kfp.json
+
+if  [[ ! -z "$JUPYTERHUB_USER_NAME" ]]; then
+  # Use COS_BUCKET env variable set in spawner, if not set, use default
+  export COS_BUCKET=${COS_BUCKET:-default}
+else
+  # Otherwise, use jupyterhub username and replace any special characters with a dash
+  COS_BUCKET=$(python -c 'import sys;print(sys.argv[1].translate ({ord(c): "-" for c in "!@#$%^&*()[]{};:,./<>?\|`~=_+"}))' "$JUPYTERHUB_USER_NAME")
+  export COS_BUCKET
+fi
+
+if [[ ! -f "$DEFAULT_RUNTIME_FILE" ]]; then
+  elyra-metadata install runtimes --schema_name=kfp \
+                                  --name=my_kfp \
+                                  --display_name="Default Kubeflow Pipeline Runtime" \
+                                  --api_endpoint=http://ml-pipeline-ui.kubeflow/pipeline \
```

Comment:
Could this KFP instance require DEX authentication?  If so, seems like `api_username` and `api_password` arguments would be necessary.

Reply:
odh's kubeflow manifests dont appear to have a kfdef for openshift and istio(w/ dex enabled) yet. Need to check with them to see if they can support moving forward